spot_rl_experiments/spot_rl/envs/gaze_env.py

```diff
@@ -42,7 +42,15 @@ def reset(self, target_obj_name, *args, **kwargs):
         cmd_id = self.spot.set_arm_joint_positions(
             positions=self.initial_arm_joint_angles, travel_time=1
         )
-        self.spot.block_until_arm_arrives(cmd_id, timeout_sec=1)
+
+        # Block until arm arrives with incremental timeout for 3 attempts
+        timeout_sec = 1.0
+        max_allowed_timeout_sec = 3.0
+        status = False
+        while status is False and timeout_sec <= max_allowed_timeout_sec:
+            status = self.spot.block_until_arm_arrives(cmd_id, timeout_sec=timeout_sec)
+            timeout_sec += 1.0
```

Comment:
During one test, resetting of arm joint state in gaze was creating error due to timeout which was set to 1sec.

Increasing the timeout_sec was making things smoother & less prone to that timeout error.

Also, the function signature of this method in qn has default timeout of 5seconds `def block_until_arm_arrives(self, cmd_id, timeout_sec=5):` but we were putting 1 sec.



So I made it an incremental timeout from 1sec to 3sec to prevent this error.
If it can succeed the arm reset within 1sec timeout, all good. If not, then it will retry for 2 sec & 3 seconds accordingly..

Reply:
As discussed offline, this bit of logic here is required at this place because we need a safety net incase reset arm didn't succeed for whatever reason.